plugins/copy-by-storage/src/index.js

```diff
@@ -0,0 +1,239 @@
+   * `shortcut` Register cut (ctr + x), copy (ctr + c) and paste (ctr + v)
+   * in the shortcut.
+   */
+  init({
+    contextMenu = true,
+    shortcut = true,
+  } = {
+    contextMenu: true,
+    shortcut: true,
+    unregisterDuplicate: true,
+  }) {
+    if (contextMenu) {
+      // Register the menus
+      this.blockCopyToStorageContextMenu();
+      this.blockPasteFromStorageContextMenu();
+    }
+
+    if (shortcut) {
+      // Unregister the default KeyboardShortcuts
+      Blockly.ShortcutRegistry.registry.unregister(
+          Blockly.ShortcutItems.names.COPY);
+      Blockly.ShortcutRegistry.registry.unregister(
+          Blockly.ShortcutItems.names.CUT);
+      Blockly.ShortcutRegistry.registry.unregister(
+          Blockly.ShortcutItems.names.PASTE);
+      // Register the KeyboardShortcuts
+      this.blockCopyToStorageShortcut();
+      this.blockCutToStorageShortcut();
+      this.blockPasteFromStorageShortcut();
+    }
+  }
+  /**
+   * Set the copy command to context menu
+   */
+  blockCopyToStorageContextMenu() {
+    /** @type {!Blockly.ContextMenuRegistry.RegistryItem} */
+    const copyToStorageOption = {
+      displayText: function() {
+        if (Blockly.Msg['COPYBYSTORAGE_COPY']) {
+          return Blockly.Msg['COPYBYSTORAGE_COPY'];
+        }
+        return 'copy';
+      },
+      preconditionFn: function(
+          /** @type {!Blockly.ContextMenuRegistry.Scope} */ scope) {
+        return 'enabled';
+      },
+      callback: function(
+          /** @type {!Blockly.ContextMenuRegistry.Scope} */ scope) {
+        const blockText = JSON.stringify(
+            convertBlockToSaveInfo(Blockly.selected));
+        localStorage.setItem('blocklyStash', blockText);
+      },
+      scopeType: Blockly.ContextMenuRegistry.ScopeType.BLOCK,
+      id: 'blockCopyToStorage',
+      weight: 0,
+    };
+    Blockly.ContextMenuRegistry.registry.register(copyToStorageOption);
+  }
+  /**
+   * Set the paste command to context menu
+   */
+  blockPasteFromStorageContextMenu() {
+    /** @type {!Blockly.ContextMenuRegistry.RegistryItem} */
+    const pasteFromStorageOption = {
+      displayText: function() {
+        if (Blockly.Msg['COPYBYSTORAGE_PASTE']) {
+          return Blockly.Msg['COPYBYSTORAGE_PASTE'];
+        }
+        return 'paste';
+      },
+      preconditionFn: function(
+          /** @type {!Blockly.ContextMenuRegistry.Scope} */ scope) {
+        const copyData = JSON.parse(localStorage.getItem('blocklyStash'));
+        if (copyData &&
+          scope.workspace.isCapacityAvailable(copyData.typeCounts)) {
+          return 'enabled';
+        }
+        return 'disabled';
+      },
+      callback: function(
+          /** @type {!Blockly.ContextMenuRegistry.Scope} */ scope) {
+        const blockText = localStorage.getItem('blocklyStash');
+        const saveInfo = JSON.parse(blockText);
+        Blockly.serialization.blocks.append(
+            saveInfo['saveInfo'], scope.workspace);
```

Comment:
I was playing around with the plugin and it looks like it doesn't undo the pasted block properly. I think to fix this we need to just add `{recordUndo: true}` as the last parameter here and in the shortcut method as well.